README.adoc

```diff
@@ -82,6 +82,64 @@ Once that is finished, the wallet registry can call
 `Wallets.registerNewWallet`, which sets the new wallet as the active wallet.
 Going forward, it will receive deposits.
 
+==== Wallet Closure
+
+Wallets can close in a few ways:
+
+* The operators notify the chain that the wallet is failing a heartbeat: `Wallets.notifyWalletHeartbeatFailed`
+* Someone notifies the chain that the wallet timed out while filling a redemption: `Wallets.notifyWalletTimedOutRedemption`
+* Someone notifies the chain that the wallet is too old: `Wallets.notifyCloseableWallet`
+* Someone notifies the chain that the wallet has too few BTC remaining: `Wallets.notifyCloseableWallet`
+
+Each of the above routes into `Wallets.moveFunds`. If the wallet has no main
+UTXO, then we can start closing it immediately via
+`Wallets.beginWalletClosing`. Otherwise, we change it's state to
+`WalletState.MovingFunds`, decrease the live wallet count, and give the wallet
+a deadline to move its funds to other wallet(s). If this was the active wallet,
+then we currently have no active wallet.
+
+After `movingFundsTimeout` goes by, anyone can call
+`MovingFunds.notifyMovingFundsTimeout` which pipes into
+`Wallets.terminateWallet` followed by `ecdsaWalletRegistry.seize`.
+`terminateWallet`, in this case passes through to
+`ecdsaWalletRegistry.closeWallet` after changing the state to
+`WalletState.Terminated`.
+
+Furthermore, the `ecdsaWalletRegistry.seize` call is punishing the operators by
+`movedFundsSweepTimeoutSlashingAmount` and rewarding the notifier with a reward
+multiplier of `movedFundsSweepTimeoutNotifierRewardMultiplier` (the notifier
+gets a percentaged of the slashed stake).
+
+In order to avoid this, the wallet has to commit to which wallets they'll send
+the funds to (`MovingFunds.submitMovingFundsCommitment`), then actually send
```

Comment:
```suggestion
the funds to (`Bridge.submitMovingFundsCommitment`), then actually send
```

Reply:
Making the change!

If `MovingFunds.submitMovingFundsCommitment` is also `external` why do we have the layer of indirection?

Reply:
`MovingFunds` is a library, `Bridge` is a contract. `MovingFunds` does not have its own storage, it's using `Bridge` storage (`BridgeState`) and we can not call `MovingFunds` directly. The fact it's `external` is just about that it's called via DELEGATECALL and the compiler is not inlining the code of the called function into `Bridge` contract so that we can save on the contract bytecode size (24 KB limit on EVM).